src/utils/utils.ts

```diff
@@ -416,22 +416,27 @@ export function createPostgresPool(
     configOrDatabaseUrl: PluginsServerConfig | string,
     onError?: (error: Error) => any
 ): Pool {
+    if (typeof configOrDatabaseUrl !== 'string') {
+        if (!configOrDatabaseUrl.DATABASE_URL && !configOrDatabaseUrl.POSTHOG_DB_NAME) {
+            throw new Error('Invalid configuration for Postgres: either DATABASE_URL or POSTHOG_DB_NAME required')
+        }
+    }
     const credentials: Partial<PoolConfig> =
         typeof configOrDatabaseUrl === 'string'
             ? {
                   connectionString: configOrDatabaseUrl,
               }
-            : configOrDatabaseUrl.POSTHOG_DB_NAME
+            : configOrDatabaseUrl.DATABASE_URL
             ? {
-                  database: configOrDatabaseUrl.POSTHOG_DB_NAME,
+                  connectionString: configOrDatabaseUrl.DATABASE_URL,
+              }
+            : {
+                  database: configOrDatabaseUrl.POSTHOG_DB_NAME || undefined,
```

Comment:
Is there a better way to convert `null` to `undefined` for types?

Reply:
You can also use `??` which is the _nullish_ coalescing operator – where `||` returns right side when left side is _falsy_, `??` returns right side only when left side is `null` or `undefined`. So depends on whether we want the plugin server to treat `''` as a valid (if empty) string value (→ `??`), or as a lack of value i.e. `undefined` (→ `||`).

Reply:
Will do what PostHog main does: null is not ok, falsy is ok.